server/src/main/java/org/elasticsearch/action/admin/cluster/migration/TransportGetFeatureUpgradeStatusAction.java

```diff
@@ -60,16 +64,50 @@ public TransportGetFeatureUpgradeStatusAction(
     @Override
     protected void masterOperation(Task task, GetFeatureUpgradeStatusRequest request, ClusterState state,
                                    ActionListener<GetFeatureUpgradeStatusResponse> listener) throws Exception {
-        List<GetFeatureUpgradeStatusResponse.IndexVersion> indexVersions = new ArrayList<>();
-        indexVersions.add(new GetFeatureUpgradeStatusResponse.IndexVersion(".security-7", "7.1.1"));
-        List<GetFeatureUpgradeStatusResponse.FeatureUpgradeStatus> features = new ArrayList<>();
-        features.add(new GetFeatureUpgradeStatusResponse.FeatureUpgradeStatus(
-            "security",
-            "7.1.1",
-            "UPGRADE_NEEDED",
+
+        List<GetFeatureUpgradeStatusResponse.FeatureUpgradeStatus> features = systemIndices.getFeatures().entrySet().stream()
+            .map(entry -> getFeatureUpgradeStatus(state, entry))
+            .collect(Collectors.toList());
+
+        boolean isUpgradeNeeded = features.stream()
+                .map(GetFeatureUpgradeStatusResponse.FeatureUpgradeStatus::getMinimumIndexVersion)
+                .min(Version::compareTo)
+                .orElse(Version.CURRENT)
+                .before(Version.V_7_0_0);
+
+        listener.onResponse(new GetFeatureUpgradeStatusResponse(features, isUpgradeNeeded ? "UPGRADE_NEEDED" : "NO_UPGRADE_NEEDED"));
```

Comment:
Nit: Can you move `"UPGRADE_NEEDED"` and `"NO_UPGRADE_NEEDED"` into a constant?

Reply:
Good call. I should have done that in the API stubs PR.

Reply:
I used an enum, but let me know if you think a pair of string constants would be better.